pkg/azure/provider/helpers/driver.go

```diff
@@ -0,0 +1,676 @@
+	vmName := *vm.Name
+	err := accesshelpers.SetCascadeDeleteForNICsAndDisks(ctx, vmAccess, resourceGroup, vm)
+	if err != nil {
+		return status.Error(codes.Internal, fmt.Sprintf("Failed to update cascade delete of associated resources for VM: [ResourceGroup: %s, Name: %s], Err: %v", resourceGroup, vmName, err))
+	}
+	err = accesshelpers.DeleteVirtualMachine(ctx, vmAccess, resourceGroup, vmName)
+	if err != nil {
+		return status.Error(codes.Internal, fmt.Sprintf("Failed to delete VM: [ResourceGroup: %s, Name: %s], Err: %v", resourceGroup, vmName, err))
+	}
+	return nil
+}
+
+func createNICDeleteTask(resourceGroup, nicName string, nicAccess *armnetwork.InterfacesClient) utils.Task {
+	return utils.Task{
+		Name: fmt.Sprintf("delete-nic-[resourceGroup: %s name: %s]", resourceGroup, nicName),
+		Fn: func(ctx context.Context) error {
+			return accesshelpers.DeleteNIC(ctx, nicAccess, resourceGroup, nicName)
+		},
+	}
+}
+
+func createDisksDeletionTask(resourceGroup string, diskNames []string, diskAccess *armcompute.DisksClient) utils.Task {
+	taskFn := func(ctx context.Context) error {
+		var errs []error
+		for _, diskName := range diskNames {
+			klog.Infof("Deleting disk: [ResourceGroup: %s, DiskName: %s]", resourceGroup, diskName)
+			if err := accesshelpers.DeleteDisk(ctx, diskAccess, resourceGroup, diskName); err != nil {
+				errs = append(errs, err)
+			}
+		}
+		return errors.Join(errs...)
+	}
+	return utils.Task{
+		Name: fmt.Sprintf("delete-disks-[resourceGroup: %s]", resourceGroup),
+		Fn:   taskFn,
+	}
+}
+
+// Helper functions for driver.CreateMachine
+// ---------------------------------------------------------------------------------------------------------------------
+
+// GetSubnet gets the subnet for the subnet configuration in the provider config.
+func GetSubnet(ctx context.Context, factory access.Factory, connectConfig access.ConnectConfig, providerSpec api.AzureProviderSpec) (*armnetwork.Subnet, error) {
+	vnetResourceGroup := providerSpec.ResourceGroup
+	if !utils.IsNilOrEmptyStringPtr(providerSpec.SubnetInfo.VnetResourceGroup) {
+		vnetResourceGroup = *providerSpec.SubnetInfo.VnetResourceGroup
+	}
+	subnetAccess, err := factory.GetSubnetAccess(connectConfig)
+	if err != nil {
+		return nil, status.WrapError(codes.Internal, fmt.Sprintf("failed to create subnet access, Err: %v", err), err)
+	}
+	subnet, err := accesshelpers.GetSubnet(ctx, subnetAccess, vnetResourceGroup, providerSpec.SubnetInfo.VnetName, providerSpec.SubnetInfo.SubnetName)
+	if err != nil {
+		return nil, status.WrapError(codes.Internal, fmt.Sprintf("failed to get subnet: [ResourceGroup: %s, Name: %s, VNetName: %s], Err: %v", vnetResourceGroup, providerSpec.SubnetInfo.SubnetName, providerSpec.SubnetInfo.VnetName, err), err)
+		//return nil, status.Error(codes.Internal, fmt.Sprintf("failed to get subnet: [ResourceGroup: %s, Name: %s, VNetName: %s], Err: %v", vnetResourceGroup, providerSpec.SubnetInfo.SubnetName, providerSpec.SubnetInfo.VnetName, err))
```

Comment:
```suggestion

```

Reply:
ah i missed removing this commented code, thanks, removed.